upgrades/2024_10_09_032151_populate_case_started.php

```diff
@@ -0,0 +1,347 @@
+            echo 'Failed to populate cases_started: ' . $e->getMessage();
+        }
+
+        echo PHP_EOL;
+    }
+
+    /**
+     * Display the processing rate of requests based on the time elapsed.
+     *
+     * @param int     $processed The number of requests that have been processed so far.
+     * @param float   $startTime The time when the processing started (in microseconds).
+     * @param int     $count     The total number of requests to be processed.
+     * @param bool    $forceShow Whether to force showing the rate, regardless of the refresh interval.
+     * @return void
+     */
+    protected function displayRate(int $processed, float $startTime, int $count, bool $forceShow): void
+    {
+        // Get the current time in microseconds
+        $currentTime = microtime(true);
+
+        // Only update the display if the forceShow flag is set or enough time has passed since the last display
+        if (!$forceShow && ($this->lastPrint + self::REFRESH_TIME > $currentTime)) {
+            return; // Skip if refresh time hasn't elapsed
+        }
+
+        // Update the last print time
+        $this->lastPrint = $currentTime;
+
+        // Calculate the time elapsed since the start of processing
+        $timeElapsed = $currentTime - $startTime;
+
+        // Calculate the processing rate (requests per second), handle divide by zero case
+        $rate = $timeElapsed > 0 ? $processed / $timeElapsed : 0;
+
+        // Clear the current line in the console
+        echo "\r";
+
+        // Display the current processing progress and rate, formatted to 2 decimal places
+        echo "    #{$processed}/{$count} Processing rate: " . number_format($rate, 2) . ' requests/second';
+    }
+
+    /**
+     * Get non-system process requests.
+     * Retrieves process requests where the parent is null
+     * and filters out system processes.
+     */
+    private function getNonSystemRequests()
+    {
+        return DB::table('process_requests')
+            ->join('processes', 'process_requests.process_id', '=', 'processes.id')
+            ->join('process_categories', 'processes.process_category_id', '=', 'process_categories.id')
+            ->whereNull('parent_request_id')
+            ->where('process_categories.is_system', false)
+            ->select(
+                'process_requests.id',
+                'process_requests.case_number',
+                'process_requests.user_id',
+                'process_requests.case_title',
+                'process_requests.case_title_formatted',
+                'process_requests.status',
+                'process_requests.initiated_at',
+                'process_requests.created_at',
+                'process_requests.completed_at',
+                'process_requests.updated_at',
+                'process_requests.name',
+                'process_requests.parent_request_id',
+                'processes.id as process_id',
+                'processes.name as process_name',
+            );
```

Comment:
Put the order by clause here
```suggestion
            )
            ->orderBy('process_requests.id');
```